### PR TITLE
Update index state to 2020-11-15

### PR DESCRIPTION
**Pull request by newhoggy**
And also add retry logic to Github Actions.

### CODE REVIEW

**Comment by newhoggy** [comment]
Some commands in Github Actions are flaky for example:

```
Run choco install -y pkgconfiglite
  choco install -y pkgconfiglite
  shell: C:\Program Files\Git\bin\bash.EXE --noprofile --norc -e -o pipefail {0}
  env:
    CABAL_BUILDDIR: dist
Chocolatey v0.10.15
Installing the following packages:
pkgconfiglite
By installing you accept licenses for the packages.

Progress: Downloading pkgconfiglite 0.28... 19%
Progress: Downloading pkgconfiglite 0.28... 61%
Progress: Downloading pkgconfiglite 0.28... 100%

pkgconfiglite v0.28
pkgconfiglite package files install completed. Performing other installation steps.
WARNING: Url has SSL/TLS available, switching to HTTPS for download
Downloading pkgconfiglite 64 bit
  from 'https://downloads.sourceforge.net/project/pkgconfiglite/0.28-1/pkg-config-lite-0.28-1_bin-win32.zip'
ERROR: The remote file either doesn't exist, is unauthorized, or is forbidden for url 'https://downloads.sourceforge.net/project/pkgconfiglite/0.28-1/pkg-config-lite-0.28-1_bin-win32.zip'. Exception calling "GetResponse" with "0" argument(s): "The operation has timed out"
  pkgconfiglite may be able to be automatically uninstalled.
The install of pkgconfiglite was NOT successful.
Error while running 'C:\ProgramData\chocolatey\lib\pkgconfiglite\tools\chocolateyInstall.ps1'.
 See log for details.

Chocolatey installed 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Failures
 - pkgconfiglite (exited 404) - Error while running 'C:\ProgramData\chocolatey\lib\pkgconfiglite\tools\chocolateyInstall.ps1'.
 See log for details.
Error: Process completed with exit code 148.
```

The retry logic will retry some of these flaky commands in the hope the second attempt will work.

File: .github/bin/retry [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
```
Comment (erikd) [inline .github/bin/retry:1]: Please specifiy in commit message header that this is about GH actions.

**Review by erikd — APPROVED** [review]
Just need to improve that commit message but otherwise :+1: .

File: .github/bin/retry (hunk 1 above)
Comment (newhoggy) [reply .github/bin/retry:1]: Modified commit message to "Add retry logic in Github Actions"

**Comment by newhoggy** [comment]
bors merge

File: cabal.project [hunk 2]
```diff
@@ -183,6 +182,7 @@ constraints:
     -- systemd-2.3.0 requires at least network 3.1.1.0 but it doesn't declare
     -- that dependency
   , network >= 3.1.1.0
+  , streaming-bytestring < 0.1.7
```
Comment (mrBliss) [inline cabal.project:185]: So 0.1.6 is fine, but 0.1.7 is not? That's not very PVP/semver :confused:

**Pull request merged**